#VRML_SIM R2023b utf8
# license: Apache License 2.0
# license url: http://www.apache.org/licenses/LICENSE-2.0
# This is a proto file for Webots for the PAI-urdf
# Extracted from: urdf/PAI-urdf.urdf

PROTO PAI-urdf [
  field  SFVec3f     translation     0 0 0
  field  SFRotation  rotation        0 0 1 0
  field  SFString    name            "PAI-urdf"  # Is `Robot.name`.
  field  SFString    controller      "void"      # Is `Robot.controller`.
  field  MFString    controllerArgs  []          # Is `Robot.controllerArgs`.
  field  SFString    customData      ""          # Is `Robot.customData`.
  field  SFBool      supervisor      FALSE       # Is `Robot.supervisor`.
  field  SFBool      synchronization TRUE        # Is `Robot.synchronization`.
  field  SFBool      selfCollision   FALSE       # Is `Robot.selfCollision`.
]
{
  Robot {
    translation IS translation
    rotation IS rotation
    controller IS controller
    controllerArgs IS controllerArgs
    customData IS customData
    supervisor IS supervisor
    synchronization IS synchronization
    selfCollision IS selfCollision
    children [
      Accelerometer {
      }
      Gyro {
      }
      Shape {
        appearance DEF base_link_material PBRAppearance {
          roughness 1
          metalness 0
        }
        geometry DEF base_link Mesh {
          url [
            "../protos/meshes/base_link.STL"
          ]
        }
      }
      HingeJoint {
        jointParameters HingeJointParameters {
          axis -1 0 0
          anchor 0 0.075 0
        }
        device [
          RotationalMotor {
            name "Roll-L"
            maxTorque 10000
          }
          PositionSensor {
            name "Roll-L_sensor"
          }
        ]
        endPoint Solid {
          translation 0 0.075 0
          rotation -1 0 0 0
          children [
            Shape {
              appearance DEF left-hip-abad_material PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry DEF left-hip-abad Mesh {
                url [
                  "../protos/meshes/left-hip-abad.STL"
                ]
              }
            }
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
                anchor 0.049 0 -0.0005
              }
              device [
                RotationalMotor {
                  name "Yaw-L"
                  maxTorque 10000
                }
                PositionSensor {
                  name "Yaw-L_sensor"
                }
              ]
              endPoint Solid {
                translation 0.049 0 -0.0005
                children [
                  Shape {
                    appearance DEF left-hip-yaw_material PBRAppearance {
                      roughness 1
                      metalness 0
                    }
                    geometry DEF left-hip-yaw Mesh {
                      url [
                        "../protos/meshes/left-hip-yaw.STL"
                      ]
                    }
                  }
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 1 0
                    }
                    device [
                      RotationalMotor {
                        name "Pitch-L"
                        maxTorque 10000
                      }
                      PositionSensor {
                        name "Pitch-L_sensor"
                      }
                    ]
                    endPoint Solid {
                      rotation 0 1 0 0
                      children [
                        Shape {
                          appearance DEF left-hip-pitch_material PBRAppearance {
                            roughness 1
                            metalness 0
                          }
                          geometry DEF left-hip-pitch Mesh {
                            url [
                              "../protos/meshes/left-hip-pitch.STL"
                            ]
                          }
                        }
                        HingeJoint {
                          jointParameters HingeJointParameters {
                            axis 0 1 0
                            anchor 0 0 -0.15
                          }
                          device [
                            RotationalMotor {
                              name "Knee-L"
                              maxTorque 10000
                            }
                            PositionSensor {
                              name "Knee-L_sensor"
                            }
                          ]
                          endPoint Solid {
                            translation 0 0 -0.15
                            rotation 0 1 0 0
                            children [
                              Shape {
                                appearance DEF left-knee_material PBRAppearance {
                                  roughness 1
                                  metalness 0
                                }
                                geometry DEF left-knee Mesh {
                                  url [
                                    "../protos/meshes/left-knee.STL"
                                  ]
                                }
                              }
                              HingeJoint {
                                jointParameters HingeJointParameters {
                                  axis 0 1 0
                                  anchor 0 0 -0.16
                                }
                                device [
                                  RotationalMotor {
                                    name "Ankle-L"
                                    maxTorque 10000
                                  }
                                  PositionSensor {
                                    name "Ankle-L_sensor"
                                  }
                                ]
                                endPoint Solid {
                                  translation 0 0 -0.16
                                  rotation 0 1 0 0
                                  children [
                                    Shape {
                                      appearance DEF left-ankle_material PBRAppearance {
                                        roughness 1
                                        metalness 0
                                      }
                                      geometry DEF left-ankle Mesh {
                                        url [
                                          "../protos/meshes/left-ankle.STL"
                                        ]
                                      }
                                    }
                                  ]
                                  name "left-ankle"
                                  boundingObject USE left-ankle
                                  physics Physics {
                                    density -1
                                    mass 0.09
                                    centerOfMass [
                                      0.010818 0 -0.014672
                                    ]
                                    inertiaMatrix [
                                      2.76e-06 2.7107e-05 2.6942e-05
                                      3.1498e-11 1.0465e-06 3.7297e-13
                                    ]
                                  }
                                }
                              }
                            ]
                            name "left-knee"
                            boundingObject USE left-knee
                            physics Physics {
                              density -1
                              mass 0.36
                              centerOfMass [
                                -0.004596 0.012568 -0.063722
                              ]
                              inertiaMatrix [
                                0.00011945 0.00012253 2.0036e-05
                                6.4077e-08 3.3225e-07 1.2844e-07
                              ]
                            }
                          }
                        }
                      ]
                      name "left-hip-pitch"
                      boundingObject USE left-hip-pitch
                      physics Physics {
                        density -1
                        mass 0.2
                        centerOfMass [
                          0 0 -0.05727
                        ]
                        inertiaMatrix [
                          0.00012318 0.00012926 6.3186e-06
                          3.9623e-20 -2.6324e-20 7.2271e-23
                        ]
                      }
                    }
                  }
                ]
                name "left-hip-yaw"
                boundingObject USE left-hip-yaw
                physics Physics {
                  density -1
                  mass 0.65
                  centerOfMass [
                    -0.000496 0 0.003295
                  ]
                  inertiaMatrix [
                    6.7757e-05 6.8581e-05 6.7914e-05
                    7.9552e-21 -6.395e-07 9.1346e-21
                  ]
                }
              }
            }
          ]
          name "left-hip-abad"
          boundingObject USE left-hip-abad
          physics Physics {
            density -1
            mass 0.35
            centerOfMass [
              0.041409 0 0.055119
            ]
            inertiaMatrix [
              3.6841e-05 3.8839e-05 3.6277e-05
              -2.584e-22 -6.2162e-07 -1.9482e-21
            ]
          }
        }
      }
      HingeJoint {
        jointParameters HingeJointParameters {
          axis -1 0 0
          anchor 0 -0.075 0
        }
        device [
          RotationalMotor {
            name "Roll-R"
            maxTorque 10000
          }
          PositionSensor {
            name "Roll-R_sensor"
          }
        ]
        endPoint Solid {
          translation 0 -0.075 0
          rotation -1 0 0 0
          children [
            Shape {
              appearance DEF right-hip-abad_material PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry DEF right-hip-abad Mesh {
                url [
                  "../protos/meshes/right-hip-abad.STL"
                ]
              }
            }
            HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 0 1
                anchor 0.049 0 -0.0005
              }
              device [
                RotationalMotor {
                  name "Yaw-R"
                  maxTorque 10000
                }
                PositionSensor {
                  name "Yaw-R_sensor"
                }
              ]
              endPoint Solid {
                translation 0.049 0 -0.0005
                children [
                  Shape {
                    appearance DEF right-hip-yaw_material PBRAppearance {
                      roughness 1
                      metalness 0
                    }
                    geometry DEF right-hip-yaw Mesh {
                      url [
                        "../protos/meshes/right-hip-yaw.STL"
                      ]
                    }
                  }
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      axis 0 -1 0
                    }
                    device [
                      RotationalMotor {
                        name "Pitch-R"
                        maxTorque 10000
                      }
                      PositionSensor {
                        name "Pitch-R_sensor"
                      }
                    ]
                    endPoint Solid {
                      rotation 0 -1 0 0
                      children [
                        Shape {
                          appearance DEF right-hip-pitch_material PBRAppearance {
                            roughness 1
                            metalness 0
                          }
                          geometry DEF right-hip-pitch Mesh {
                            url [
                              "../protos/meshes/right-hip-pitch.STL"
                            ]
                          }
                        }
                        HingeJoint {
                          jointParameters HingeJointParameters {
                            axis 0 -1 0
                            anchor 0 0 -0.15
                          }
                          device [
                            RotationalMotor {
                              name "Knee-R"
                              maxTorque 10000
                            }
                            PositionSensor {
                              name "Knee-R_sensor"
                            }
                          ]
                          endPoint Solid {
                            translation 0 0 -0.15
                            rotation 0 -1 0 0
                            children [
                              Shape {
                                appearance DEF right-knee_material PBRAppearance {
                                  roughness 1
                                  metalness 0
                                }
                                geometry DEF right-knee Mesh {
                                  url [
                                    "../protos/meshes/right-knee.STL"
                                  ]
                                }
                              }
                              HingeJoint {
                                jointParameters HingeJointParameters {
                                  axis 0 -1 0
                                  anchor 0 0 -0.16
                                }
                                device [
                                  RotationalMotor {
                                    name "Ankle-R"
                                    maxTorque 10000
                                  }
                                  PositionSensor {
                                    name "Ankle-R_sensor"
                                  }
                                ]
                                endPoint Solid {
                                  translation 0 0 -0.16
                                  rotation 0 -1 0 0
                                  children [
                                    Shape {
                                      appearance DEF right-ankle_material PBRAppearance {
                                        roughness 1
                                        metalness 0
                                      }
                                      geometry DEF right-ankle Mesh {
                                        url [
                                          "../protos/meshes/right-ankle.STL"
                                        ]
                                      }
                                    }
                                  ]
                                  name "right-ankle"
                                  boundingObject USE right-ankle
                                  physics Physics {
                                    density -1
                                    mass 0.09
                                    centerOfMass [
                                      0.010818 0 -0.014672
                                    ]
                                    inertiaMatrix [
                                      2.76e-06 2.7107e-05 2.6942e-05
                                      3.1498e-11 1.0465e-06 3.7297e-13
                                    ]
                                  }
                                }
                              }
                            ]
                            name "right-knee"
                            boundingObject USE right-knee
                            physics Physics {
                              density -1
                              mass 0.36
                              centerOfMass [
                                -0.004596 -0.012568 -0.063722
                              ]
                              inertiaMatrix [
                                0.00011945 0.00012253 2.0036e-05
                                -6.4077e-08 3.3225e-07 -1.2844e-07
                              ]
                            }
                          }
                        }
                      ]
                      name "right-hip-pitch"
                      boundingObject USE right-hip-pitch
                      physics Physics {
                        density -1
                        mass 0.2
                        centerOfMass [
                          0 0 -0.05727
                        ]
                        inertiaMatrix [
                          0.00012318 0.00012926 6.3186e-06
                          3.9681e-20 -3.2905e-21 2.887e-20
                        ]
                      }
                    }
                  }
                ]
                name "right-hip-yaw"
                boundingObject USE right-hip-yaw
                physics Physics {
                  density -1
                  mass 0.65
                  centerOfMass [
                    -0.000496 0 0.003295
                  ]
                  inertiaMatrix [
                    6.7757e-05 6.8581e-05 6.7914e-05
                    -1.4631e-21 -6.395e-07 1.1283e-20
                  ]
                }
              }
            }
          ]
          name "right-hip-abad"
          boundingObject USE right-hip-abad
          physics Physics {
            density -1
            mass 0.35
            centerOfMass [
              0.041409 0 0.055119
            ]
            inertiaMatrix [
              3.6841e-05 3.8839e-05 3.6277e-05
              2.4386e-22 -6.2162e-07 -2.348e-21
            ]
          }
        }
      }
    ]
    name IS name
    boundingObject USE base_link
    physics Physics {
      density -1
      mass 1.500000
      centerOfMass [ 0.069429 0.000000 0.093307 ]
      inertiaMatrix [
        2.878700e-03 4.760600e-03 4.302900e-03
        -6.202800e-19 1.149600e-03 5.627800e-19
      ]
    }
  }
}
